common/app/templates/headerInlineJS/loadFonts.scala.js

```diff
@@ -1,87 +1,157 @@
+                            } else {
+                                // All others use cleartype
+                                hinting = 'Cleartype';
+                            }
+                        }
+                    }
+                } catch (e) {}
+            }
+            return hinting;
+        };
+        return determineHinting;
+    })();
 
     // Load fonts from `localStorage`.
     function loadFontsFromStorage() {
-        var storedFontFormat = localStorage['gu.fonts.format'],
-            fonts, fontFormat, font, dataAttrName, nameAndCacheKey, fontData;
-
-        if (storedFontFormat) {
-            fonts = document.querySelectorAll('.webfont');
-            fontFormat = JSON.parse(storedFontFormat).value;
-
-            for (var i = 0, j = fonts.length; i < j; ++i) {
-                font = fonts[i];
-                dataAttrName = 'data-cache-file-' + (fontHinting === 'Off' ? '' : 'hinted-' + fontHinting + '-') + fontFormat;
-                nameAndCacheKey = font.getAttribute(dataAttrName).match(/fonts\/([^/]*?)\/?([^/]*)\.(woff2|woff|tff).json$/);
-                fontData = localStorage['gu.fonts.' + nameAndCacheKey[2] + '.' + nameAndCacheKey[1]];
-
-                if (fontData) {
-                    font.innerHTML = JSON.parse(fontData).value;
-                    font.setAttribute('data-loaded-from', 'local');
+        if ("localStorage" in window) {
+            try {
+                // detect which font format (ttf, woff, woff2 etc) we want
+                var fontFormat = (function () {
+                    var fontFormat = localStorage.getItem('gu.fonts.format');
+
+                    function getFontFormat() {
+                        if (!fontFormat) {
+                            function supportsWoff2() {
+                                if ("FontFace" in window) { // https://github.com/filamentgroup/woff2-feature-test
+                                    var f = new window.FontFace('t', 'url("data:application/font-woff2,") format("woff2")', {});
+                                    f.load().catch(function(){});
+                                    return f.status === 'loading';
+                                } else if (!/edge\/([0-9]+)/.test(ua)) { // don't let edge tell you it's chrome when it's not
+                                    var browser = /(chrome|firefox)\/([0-9]+)/.exec(ua),
+                                        supportsWoff2 = {
+                                            'chrome': 36,
+                                            'firefox': 39
+                                        };
+                                    return !!browser && supportsWoff2[browser[1]] < parseInt(browser[2], 10);
+                                }
+                                return false;
+                            };
+                            fontFormat = localStorage['gu.fonts.format'] = supportsWoff2() ? 'woff2' : ua.indexOf('android') > -1 ? 'ttf' : 'woff';
```

Comment:
I prefer `localStorage.setItem()` instead of `localStorage['gu.fonts.format'] =`